### DISCUSSION TITLE
Problems safety door enable

### DISCUSSION THREAD

**Discussion by keverin16** [grblHAL/RP2040#72] [General]
Hello, i can build Grblhal for rp2040 without problems on a raspberry 400. But as soon as i uncomment "#define SAFETY_DOOR_ENABLE" in "my_machine.h" it does not build correctly and finishes with exit code 2. I would like to use the safety door input so, can somebody please help me? thank you very much!

**Comment by terjeio** [comment]
Fix committed. Thanks for reporting.

**Comment by keverin16** [comment]
ok thanks but now it doesnt work for me anymore, even if i didnt change anything. i always get many errors while building with SPI pins undeclared errors. around 12. "did you mean SPIPORT?" what could the problem be?

**Reply by keverin16** [reply]
I mean the build doesnt finishes and exits with exit code 2 and 12 "Problems". all sound more or less the same and are about spiSPI_PORTs that are undeclared.

**Reply by terjeio** [reply]
Submodules are [out of sync](https://github.com/grblHAL/core/wiki/Compiling-grblHAL#getting-the-source-code)?
Which board are you compiling for?
Post a few of the errors you get.

**Reply by keverin16** [reply]
For now i just build it likes it comes after downloading the full project from github. It should then compile for "generic_map.h" right? Last time when i did that in june it worked. Sorry i am not a developer in any form. Its a bit new for me :D thank you for your help! Here are the errors i get:

spi.c|Ln 30, Col 20]: in expansion of macro 'sPlport!
spi.c[ln 37, Col 17]: in expansion of macro 'sPiport
spi.c[Ln 82, Col 18] in expansion of macro 'SPIPORT
3 spiSPL_ PORT undeclared (first use in this function), did you mean 'SPIPORT? GCC [n 31,Col 21]
spi.c[Ln 30, Col 20|: in expansion of macro 'SPlport
spi.c|Ln 37, Col 17] in expansion of macro 'SPlport
spi.c|[un 115, Col 251 in expansion of macro 'SPIPORT
spisPL PORT undeclared (irst use in this function); did you mean 'SPIPORT? GCC [Ln 31,Col 21]
spi.c[un 30, Col 20]: in expansion of macro 'SPlportl"
spi.c[Ln 37, Col 17]: in expansion of macro `SPiport
spi.c[un 124, Col 20]: in expansion of macro 'SPIPORT
○ 'spisPL_ PORT undeclared (first use in this function); did you mean SPIPORT? GCC [Ln31,Col21]
spi.c[n 30, Col 20]: in expansion of macro `sPiportr
spi.c[un 37, Col 17] in expansion of macro 'SPlport
spi.c[ln 131, Col 21] in expansion of macro 'SPIPORT
SpiSPL PORT undeclared (first use in this function); did you mean 'SPIPORT? GCC [Ln31, Col 21]
spi.c[un 30, Col 20]: in expansion of macro 'SPlportr
spi.c[un 37, Col 17] in expansion of macro 'SPiport'
spi.c|un 137, Col 25] in expansion of macro 'SPIPORT
'spisPl PORT undeclared (first use in this function); did you mean 'SPIPORT? GCC [Ln 31, Col 21]
spi.c[Ln 30, Col 20]: in expansion of macro 'SPiportr
spi.c[Ln 37, Col 171 in expansion of macro 'SPlport
spi.c[un 153, Col 24] in expansion of macro 'SPIPORT
○ 'DREO_SPISPIPORT_RX undeclared (first use in this function); did you mean 'DREQ_SPI1_TX? GCC [Ln 33,Col 23]
spi.c[Ln 32, Col 22] in expansion of macro 'SPIdregrxl
spi.c[un 38, Col 181 in expansion of macro 'SPIdregRX
spi.c[ln 95, Col 49] in expansion of macro 'SPIDMARX
C 'DREQ_ SPISPI PORT_TX undeclared (first use in this function); did you mean 'DREQ_SPI1_TX'? GCC [Ln 35,Col 23]
spi.c|Ln 34, Col 22 in expansion of macro 'sPidreqtf
spi.c/un 39, Col 18) in expansion of macro 'sPldregTX
spi.c/Ln 103, Col 49]; in expansion of macro 'SPIDMATX
"SPI_SCK_ PIN undeclared (first use in this function) Gcc [Ln 60,Co| 16]
spi.c[n 60, Col 16): each undeclared identifier is reported only once for each function it appears in
O "SPI_MISO_PIN undeclared (first use in this function): did you mean 'SPI_MSB_FIRST? GCC [Ln 67, Col 16]

**Reply by terjeio** [reply]
My bad, I left [ethernet on in CMakeLists.txt](https://github.com/grblHAL/RP2040/blob/2ad3fb990cbb8491f5a349757e858e14014aa25d/CMakeLists.txt#L6C1-L6C7) when committing. Change to `OFF`.

**Reply by keverin16** [reply]
Sorry i forgot to thank you. its all working now, thanks!!
Btw one small question:
- Is there a setting somewhere that allows in idle with door open to jog the machine? But the spindle has to stay off.

**Reply by terjeio** [reply]
> Is there a setting somewhere that allows in idle with door open to jog the machine? But the spindle has to stay off.

`$61=1` should allow that.

**Comment by terjeio** [comment]
Fix committed.
